€(asctime)s €(levelname)s: €(message)s [in €pathnames:1891]
Traceback (most recent call last):
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\main\routes.py", line 21, in index
    user_announcements = Announcement.query.filter(or_(Announcement.academy_id==current_user.academy_id, Announcement.for_all==True)).order_by(Announcement.datetime.asc()).paginate(page, current_app.config['ITEMS_PER_PAGE'], False)
UnboundLocalError: local variable 'page' referenced before assignment
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:1891]
Traceback (most recent call last):
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\main\routes.py", line 23, in index
    if groups.has_next else None
NameError: name 'groups' is not defined
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:1891]
Traceback (most recent call last):
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\main\routes.py", line 26, in index
    return render_template('index.html', title='Home', announcements=user_announcements, next_url=next_url, prev_url=prev_url)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\templating.py", line 137, in render_template
    return _render(
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\jinja2\environment.py", line 1090, in render
    self.environment.handle_exception()
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\jinja2\environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\jinja2\_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\templates\index.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block "doc"
    {%- block html %}
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block "html"
    {% block body -%}
  File "c:\users\kwssw\onedrive\documentos\workspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block "body"
    {% block content -%}
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\templates\base.html", line 65, in block "content"
    {% block app_content %}{% endblock %}
  File "C:\Users\kwssw\OneDrive\Documentos\Workspace\academy\app\templates\index.html", line 11, in block "app_content"
    {% for i in announcements %}
TypeError: 'Pagination' object is not iterable
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
